src/resolve/resolved-conf.c

```diff
@@ -227,6 +228,105 @@ int config_parse_search_domains(
         return 0;
 }
 
+int config_parse_dns_netservice_name(const char *unit, const char *filename, unsigned line, const char *section, unsigned section_line, const char *lvalue, int ltype, const char *rvalue, void *data, void *userdata) {
+        DnsNetservice *ns = userdata;
+        _cleanup_free_ char *instance_name = NULL;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(ns);
+
+        if (isempty(rvalue)) {
+                r = -EINVAL;
+                log_syntax(unit, LOG_ERR, filename, line, r, "Service instance name can't be empty. Ignoring.");
+                return r;
+        }
+
+        if (strstr(rvalue, "%h")) {
+                _cleanup_free_ char *hostname = NULL;
+
+                r = gethostname_strict(&hostname);
+                if (r < 0) {
+                        log_warning_errno(r, "Can't get hostname for network service: %m. Ignoring.");
+                        return r;
+                }
+                instance_name = strreplace(rvalue, "%h", hostname);
+        } else
+                instance_name = strdup(rvalue);
+
+        if (!instance_name)
+                return log_oom();
+
+        /* Instance names can't be longer than 63 octets. See RFC6763 p.4.4.1. */
+        if (strlen(instance_name) > 63) {
+                r = -EINVAL;
+                log_syntax(unit, LOG_ERR, filename, line, r, "Service instance name '%s' is too long. Ignoring.", instance_name);
```

Comment:
so, we usually only include error codes in our log messages that we receive, not the ones we generate. I.e. since we *generate* EINVAL here, you should pass `0` to log_syntax instead of `r` as error code. After all the error will show up in ERRNO= in the journal...